### PR TITLE
add types to app hook

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- Adds typechecking to the app hook
- This is necessary pre-work for using the app hook in the refactored root component (separate PR)

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/53446/in-helm-managed-mode-user-is-prompted-to-upload-license
#### Does this PR introduce a user-facing change?
NONE- refactor 

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: web/README.md [hunk 1]
```diff
@@ -14,6 +14,15 @@ When a git tag is made, tagged images (not :alpha) are created.
...
 
+### All CI tests
+
+```
+yarn test
+
+```
+
+** This will run all of the `test:*` commands concurrently. Recommend using to determine if there is an error. Use the individual commands for debugging since the logging output is easier to read than the output from `concurrently`.
+
```
Comment (rianfowler) [inline web/README.md:25]: updated the scripts.

File: web/src/features/App/api/getApps.tsx [hunk 2]
```diff
@@ -1,16 +1,21 @@
 // This hook has not been integrated yet.
-import { useQuery } from "react-query";
+import React from "react";
+import { useQuery, UseQueryResult } from "react-query";
 import { Utilities } from "../../../utilities/utilities";
 
+export type App = {
```
[the remark this answers is not in the document]
Comment (miaawong) [reply web/src/features/App/api/getApps.tsx:6]: i think App is fine

File: web/src/features/App/api/getApps.tsx [hunk 3]
```diff
@@ -24,19 +29,37 @@ async function getApps({
...
+function UseApps({
+  children,
+}: {
+  children: (
+    props: UseQueryResult<{ apps: App[] | null }, Error>
+  ) => React.ReactNode;
+}) {
   const query = useApps();
 
+  // TODO: figure this out
```
Comment (miaawong) [inline web/src/features/App/api/getApps.tsx:61]:
```suggestion
  // TODO: figure this out
  // useApps() allows us to use this hook in class component
```

File: web/src/features/AppVersionHistory/hooks/useCurrentApp.tsx [hunk 4]
```diff
@@ -1,19 +1,19 @@
 // This hook has not been integrated yet. It's considered a work in progress
 import { useParams } from "react-router-dom";
 import { useState, useEffect } from "react";
-import { useApps } from "../api/getApps";
+import { App, useApps } from "@features/App";
```
[the remark this answers is not in the document]
Comment (miaawong) [reply web/src/features/AppVersionHistory/hooks/useCurrentApp.tsx:4]: great! 
I believe it's better to put into `/hooks` folder. 
from bulletproof: `/hooks` shared hooks used across the entire application

File: web/src/features/AppVersionHistory/hooks/useCurrentApp.tsx [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-  const { apps } = data;
+  const { apps = [] } = data || {};
 
-  const [currentApp, setCurrentApp] = useState(null);
+  const [currentApp, setCurrentApp] = useState<App | null>(null);
 
   useEffect(() => {
     if (apps && isFetched) {
-      setCurrentApp(apps.find((app) => app.slug === slug));
+      setCurrentApp(apps.find((app: App) => app.slug === slug) || null);
```
[the remark this answers is not in the document]
Comment (miaawong) [reply web/src/features/AppVersionHistory/hooks/useCurrentApp.tsx:16]: nice! I'll do that in the future!

**Review by miaawong — APPROVED** [review]
amazing work!

**Pull request merged**